src/dra/75F54.c

```diff
@@ -400,7 +400,365 @@ s32 CheckWingSmashInput(void) {
+            PLAYER.palette = 0x810D;
+            D_800AFDA6[0] = (s16)PLAYER.animCurFrame;
+            if (g_Player.unk66 == 1) {
+                return;
+            }
+            if (g_Player.unk66 == 2) {
+                PLAYER.animSet = 0xD;
+                PLAYER.unk5A = 0;
+                D_800AFDA6[0] = 6;
+                return;
+            }
+        }
+        func_8010DA48(0xC3);
+        PLAYER.animFrameDuration = 1;
+        PLAYER.animFrameIdx = 2;
+        PLAYER.palette = 0x8100;
+        func_8010E27C();
+        PLAYER.step_s++;
+        break;
+    case 1:
+        if ((directionsPressed != 0) && (pressingCross == 0)) {
+            if (PLAYER.ext.player.unkAC == 0xC3) {
+                PLAYER.animFrameIdx /= 3;
+            }
+            PLAYER.step_s += 1;
+        } else {
+            func_8011690C(0);
+            func_8010E1EC(0x1200);
+            func_8010E234(0x1200);
+            break;
+        }
+    case 2:
+        // If you're pressing cross, you can't move and inputs are ignored.
+        if (pressingCross) {
+            directionsPressed = 0;
+        }
+        switch (directionsPressed) {
+        case 0:
+        default:
+            func_8010DA48(0xC3);
+            PLAYER.step_s = 1;
+            break;
+        case PAD_UP:
+            PLAYER.ext.player.unkAC = 0xC2;
+            if (PLAYER.velocityY < FIX(-1.25)) {
+                func_8010E234(0x1200);
+            } else {
+                PLAYER.velocityY = FIX(-1.25);
+            }
+            func_8011690C(-0x80);
+            func_8010E1EC(0x1200);
+            break;
+        case PAD_DOWN:
+            if (!(g_Player.pl_vram_flag & 1)) {
+                PLAYER.ext.player.unkAC = 0xC5;
+            } else {
+                PLAYER.ext.player.unkAC = 0xC4;
+            }
+            if (PLAYER.velocityY > FIX(1.25)) {
+                func_8010E234(0x1200);
+            } else {
+                PLAYER.velocityY = FIX(1.25);
+            }
+            func_8011690C(0);
+            func_8010E1EC(0x1200);
+            break;
+        case PAD_RIGHT:
+            PLAYER.ext.player.unkAC = 0xC2;
+            PLAYER.facingLeft = 0;
+            func_8011690C(0x180);
+            if (PLAYER.velocityX > FIX(1.25)) {
+                func_8010E1EC(0x1200);
+            } else {
+                PLAYER.velocityX = FIX(1.25);
+            }
+            func_8010E234(0x1200);
+            var_s3 = 1;
+            if (D_80138000 == 0) {
+                PlaySfx(0x64E);
+            }
+            break;
+        case PAD_LEFT:
+            PLAYER.ext.player.unkAC = 0xC2;
+            PLAYER.facingLeft = 1;
+            func_8011690C(0x180);
+            if (PLAYER.velocityX < FIX(-1.25)) {
+                func_8010E1EC(0x1200);
+            } else {
+                PLAYER.velocityX = FIX(-1.25);
+            }
+            func_8010E234(0x1200);
+            var_s3 = 1;
+            if (D_80138000 == 0) {
```

Comment:
I think this could be `g_BatNoScreetch` or something. At the beginning of the function, var_s3 is set to 0. When moving left and right, var_s3 is set to 1. Then at the end of the function that is assigned to `D_80138000`. So if we are moving left or right, make it so we don't screech, otherwise screech?

Reply:
Sounds good, I'll rename the local and global vars. How do we feel about ControlBatForm for this function?